documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
+## Setup Append - Request
+1.  `RequestId`: Long (8 bytes). (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  `writerId`: UUID (16 bytes) identifies the requesting appender.
+3.  Segment to append: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+4. `delegationToken`: This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-Appending
----------
+## Append Setup - Reply
 
-### Setup Append - Request
+1.  `RequestId`: 8 byte Long. (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  Segment to append: String (2 byte length, followed by that many bytes of Java's Modified UTF-8).
+3.  `writerId`: UUID (16 bytes) identifies the requesting appender.
+4.  `lastEventNumber`: Long (8 bytes). Specifies the last event number in the Stream.
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## AppendBlock - Request
 
-### Append Setup - Reply
+1. `writerId`: UUID (16 bytes) identifies the requesting appender.
+2. `Data`: This holds the contents of the block.
 
-1.  Segment that can be appended to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  ConnectionId - UUID (16 bytes) Identifies the requesting appender.
-3.  ConnectionOffsetAckLevel - Long (8 bytes) What was the last offset received and stored on this segment for this connectionId (0 if new)
+## AppendBlockEnd - Request
 
-### BeginAppendBlock - Request 
+1. `writerId`: UUID (16 bytes) identifies the requesting appender.
+2. `sizeOfWholeEvents`: Integer (4 bytes).
+3. `Data`: This holds the contents of the block.
+4. `numEvents`: Integer (4 bytes). Specifies the current number of events.
+5. `lastEventNumber`: Long (8 bytes). Specifies the value of last event number in the Stream.
 
-Only valid after SetupAppend has already been done successfully.
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection to this segment
-3.  Length of data before EndAppendBlock message - Integer (4 bytes) 
+## Partial Event - Request/Reply
 
-### EndAppendBlock- Request
+-  **Data**: A partial message is the one that was broken up when the message was sent over the wire. It is essential that for any reason, the whole message is reconstructed by reading the partial message in sequence and reassembled as whole message. It is not valid to attempt to start a new partial message before completing the previous one.
```

Comment:
This text is not accurate. A Partial event is an event at the end of a append block that did not fully fit in the append block. The remainder of the event will be in the appendBlockEnd.

Reply:
@tkaitchuck Replaced with the above content, Please check